site/app/templates/course/CourseMaterials.twig

```diff
@@ -131,6 +136,16 @@
         }
     }
 
+    function shareAllToOther(id) {
+        // pass folderid to the function
+        // this will find all elements with the classname same as the id and click their checkboxes
+        // this recursively sets the permissions for all files and folders
+
+        $('.'+id).each(function(){
```

Comment:
This code should keep track of which check marks are marked first. Currently it only changes each child box into the opposite state.

Reply:
Will push fix soon! :)